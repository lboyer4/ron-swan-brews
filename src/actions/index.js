export const setBreweries = (breweries) => ({
	type: 'SET_BREWERIES',
	breweries
})	