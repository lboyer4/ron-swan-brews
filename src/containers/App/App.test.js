import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps, mapStateToProps } from './index.js';
import { setBreweries, setQuotes } from '../../actions';

const noFavoriteBreweries = {	
	name: 'NOMNOM',
	brewery_type: 'micro',
	id: 1,
	city: 'Denver',
	street: 'market st.',
	phone: '800-555-5555'
}

const favoriteBreweries = {	
	name: 'NOMNOM',
	brewery_type: 'micro',
	id: 1,
	city: 'Denver',
	street: 'market st.',
	phone: '800-555-5555',
	favorited: false
}

const mockAddFavorite = jest.fn()

describe('App', () => {
	let wrapper;
	let mockSetBreweries;
	let mockSetQuotes;
	let mockAddFavorite

	beforeEach (() => {
		mockSetBreweries = jest.fn();
		mockSetQuotes = jest.fn();
		mockAddFavorite = jest.fn()

		wrapper = shallow(
			<App 
				setBreweries={mockSetBreweries}
				setQuotes={mockSetQuotes}
			/>
		)
	});

	it('should match snapshots', () => {
		expect(wrapper).toMatchSnapshot();
	});

	describe('handleBreweries', () => {

		it.skip('should fetch data with the correct params', () => {
			let mockUrl = 'https://api.openbrewerydb.org/breweries?by_state=colorado'
			const spy = jest.spyOn(App, 'fetchData');
			const wrapper = shallow(<App />)
			const instance= wrapper.instance();
			instance.handleBreweries();

			expect(spy).toHaveBeenCalled()
		})

		it.skip('should return breweries with correct props', () => {
			const noFavoriteBreweries = {	
	name: 'NOMNOM',
	brewery_type: 'micro',
	id: 1,
	city: 'Denver',
	street: 'market st.',
	phone: '800-555-5555'
}
			// wrapper.instance().addFavorite([noFavoriteBreweries])
			expect(wrapper.instance().mockAddFavorite).toEqual([favoriteBreweries])
		})

		it.skip('should call setBreweries dispatch', () => {
			wrapper.instance().addFavorite()
			expect(wrapper.instance().props.setBreweries).toHaveBeenCalled
		});
	});

	describe('handleQuotes', () => {

		it.skip('should fetch data with the correct params', () => {
		})

		it('should call setQuotes dispatch', () => {
			wrapper.instance().handleQuotes()
			expect(wrapper.instance().props.handleQuotes).toHaveBeenCalled
		});
	});

	describe('mapDispatchToProps', () => {
		it('should call setBreweries from MDTP', () => {
			const mockDispatch = jest.fn();
			const actionToDispatch = setBreweries([{brewery: 'name'}])
			const mappedProps = mapDispatchToProps(mockDispatch)

			mappedProps.setBreweries([{brewery: 'name'}])
			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
		})

		it('should call setQuotes from MDTP', () => {
			const mockDispatch = jest.fn();
			const actionToDispatch = setQuotes(['quote'])
			const mappedProps = mapDispatchToProps(mockDispatch)

			mappedProps.setQuotes(['quote'])
			expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
		});
	});
})
