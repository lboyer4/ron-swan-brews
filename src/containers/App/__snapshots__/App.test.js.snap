// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshots 1`] = `ShallowWrapper {}`;
